### PR TITLE
NEW: Management page for syncing domains that do not exist in local db

### PR DESCRIPTION
**Pull request by ilhanmercan**
OPT: Allow domain sync for domains with more than 4 nameservers

### CODE REVIEW

File: src/automats/domain_refresher.py [hunk 1]
```diff
@@ -637,7 +637,8 @@ def doDBCheckCreateUserAccount(self, *args, **kwargs):
             self.new_domain_contacts.pop('admin', None)
             self.new_domain_contacts.pop('billing', None)
             self.new_domain_contacts.pop('tech', None)
-        self.target_domain.refresh_from_db()
+        if self.target_domain:
+            self.target_domain.refresh_from_db()
```
Comment (vesellov) [inline src/automats/domain_refresher.py:641]: :+1:

File: src/back/models/domain.py [hunk 2]
```diff
@@ -142,7 +142,8 @@ def get_nameserver(self, pos):
             return self.nameserver3
         if pos == 3:
             return self.nameserver4
-        raise ValueError('Invalid position for nameserver')
+        if pos > 3:
+            return
```
Comment (vesellov) [inline src/back/models/domain.py:146]: may be at least print a warning in the logs?

**Review by vesellov — DISMISSED** [review]

File: src/back/models/domain.py (hunk 2 above)
Comment (ilhanmercan) [reply src/back/models/domain.py:146]: yes, good point

File: src/back/models/domain.py [hunk 3]
```diff
@@ -142,7 +142,9 @@ def get_nameserver(self, pos):
             return self.nameserver3
         if pos == 3:
             return self.nameserver4
-        raise ValueError('Invalid position for nameserver')
+        if pos > 3:
+            logger.warning(f'Invalid position for nameserver, position: {pos}')
```
Comment (vesellov) [inline src/back/models/domain.py:146]: @ilhanmercan please lets always use lower case in the logging to be consistent across the code  :pray:

File: src/board/views.py [hunk 4]
```diff
@@ -38,3 +44,32 @@ def form_valid(self, form):
...
+            log_events=True,
+            log_transitions=True,
+        )
+
+        domain_in_db = zdomains.domain_find(domain_name=domain_name)
+        if domain_in_db:
+            logger.info(f'{domain_name} is successfully synced.')
+            messages.success(self.request, f'{domain_name} is successfully synced.')
+        else:
+            messages.warning(self.request, f'Something went wrong during sync of {domain_name}')
```
Comment (vesellov) [inline src/board/views.py:73]: also here pls use lower case

File: src/back/models/domain.py (hunk 3 above)
Comment (ilhanmercan) [reply src/back/models/domain.py:146]: sure, changed!

File: src/board/views.py (hunk 4 above)
Comment (ilhanmercan) [reply src/board/views.py:73]: but this is not logging, a message that we return to the user in our frontend.

**Review by vesellov — APPROVED** [review]

**Pull request merged**